### PR TITLE
Consistent naming in command processors

### LINKED ISSUE #15077: Use consistent naming for command processors
## Description

The naming of processors is not consistent at the moment. This is a minor annoyance as it makes it harder to find the processors when searching for their name.

We should align rename the processors to adhere to a specific pattern. The team decided that the pattern should be: `<ValueType><Intent>Processor`. As an example, this means that the `TriggerTimerProcessor` should be renamed to `TimerTriggerProcessor`.

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This PR gives our command processor consistent names. All processors will follow the pattern `<ValueType><Intent>Processor`.

ℹ️ I am back porting this change to older versions. Not because it is required, but because it would prevent conflicts in future back ports.

## Related issues

closes #15077

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by korthout — APPROVED** [review]
Nice stuff @remcowesterhoud  🧹 

👍 Love the changes

❓ Do we also want to apply this to related topics? For example, what about `BroadcastSignalTest` and `UpdateVariableDocumentTest`? Feel free to keep this out of scope.

LGTM 👍

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCommandHandlers.java [hunk 1]
```diff
@@ -17,7 +17,7 @@ public final class ProcessInstanceCommandHandlers {
       new HashMap<>();
 
   public ProcessInstanceCommandHandlers() {
-    handlers.put(ProcessInstanceIntent.CANCEL, new CancelProcessInstanceHandler());
+    handlers.put(ProcessInstanceIntent.CANCEL, new ProcessInstanceCancelProcessor());
```
[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCommandHandlers.java:20]: Not sure what you meant by this. But looking deeper into the code of these handlers it's completely unnecessary to have them at all. All it does it "handle" the PI Cancel command. For some reason we have some vague handler here to make it inconsistent with the rest of our processors.

I will remove this entirely and make the `ProcessInstanceCancelProcessor` implement a `TypedRecordProcessor` and have it work the same way as all of our other processors.

**Comment by remcowesterhoud** [comment]
> Do we also want to apply this to related topics? For example, what about BroadcastSignalTest and UpdateVariableDocumentTest? Feel free to keep this out of scope.

I wasn't sure tbh. I think for tests it reads quite nice. I'll leave it out of scope for now and we can always decide to rename them later.

**Review by korthout — APPROVED** [review]
😍 Love the additional cleanup 🧹 

LGTM 👍

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**